src/fides/connectors/okta.py

```diff
@@ -1,102 +1,120 @@
+    """
+    Validates okta credentials by attempting to list applications with limit of 1.
 
+    Args:
+        okta_config: Configuration with OAuth2 credentials
 
-@handle_common_okta_errors
-async def validate_credentials(okta_config: Optional[OktaConfig]) -> None:
+    Raises:
+        ConnectorAuthFailureException: If authentication fails
+        ConnectorFailureException: If the API request fails for other reasons
     """
-    Calls the list_applications api with a page limit of 1 to validate
-    okta credentials.
+    try:
+        client = get_okta_client(okta_config=okta_config)
+        client.list_applications(limit=1)
+    except ConnectionException as e:
+        error_str = str(e).lower()
+        if "invalid_client" in error_str or "unauthorized" in error_str:
+            raise ConnectorAuthFailureException(
+                "Authentication failed. Please verify your OAuth2 credentials."
+            ) from e
+        raise ConnectorFailureException(f"Okta API error: {e}") from e
+
+
+def list_okta_applications(okta_client: OktaHttpClient) -> List[OktaApplication]:
     """
-    client = get_okta_client(okta_config=okta_config)
-    query_parameters = {"limit": "1"}
-    await client.list_applications(query_parameters)
+    Returns a list of Okta applications using pagination.
 
+    Args:
+        okta_client: Configured OktaHttpClient instance
 
-@handle_common_okta_errors
-async def list_okta_applications(okta_client: OktaClient) -> List[OktaApplication]:
-    """
-    Returns a list of Okta applications. Iterates through each page returned by
-    the client.
+    Returns:
+        List of OktaApplication objects from Okta API
+
+    Raises:
+        ConnectorFailureException: If the API request fails
     """
-    applications = []
-    current_applications, resp, _ = await okta_client.list_applications()
-    while True:
-        applications.extend(current_applications)
-        if resp.has_next():
-            current_applications, _ = await resp.next()
-        else:
-            break
-    return applications
+    try:
+        return okta_client.list_all_applications()
+    except ConnectionException as e:
+        raise ConnectorFailureException(f"Failed to list Okta applications: {e}") from e
 
 
 def create_okta_systems(
     okta_applications: List[OktaApplication], organization_key: str
 ) -> List[System]:
     """
-    Returns a list of system objects given a list of Okta applications
+    Returns a list of system objects given a list of Okta applications.
+
+    Only includes ACTIVE applications.
+
+    Args:
+        okta_applications: List of OktaApplication objects from Okta API
+        organization_key: The organization fides_key to associate with systems
+
+    Returns:
+        List of System objects for active Okta applications
     """
     systems = [
         System(
-            fides_key=application.id,
-            name=application.name,
+            fides_key=app.get("id", ""),
```

Comment:
I thought this was a Pydantic model now?

Reply:
yep.